landoapi/transplants.py

```diff
@@ -495,43 +606,330 @@ def check_landing_blockers(
+    blocker_open_ancestor,
+]
+
+WARNING_LINTS = [
+    warning_blocking_reviews,
+    warning_previously_landed,
+    warning_not_accepted,
+    warning_reviews_not_current,
+    warning_revision_secure,
+    warning_revision_missing_testing_tag,
+    warning_diff_warning,
+    warning_wip_commit_message,
+    warning_code_freeze,
+    warning_unresolved_comments,
+]
+
+
+def check_landing_lints(
+    transplant_state: TransplantAssessmentState,
+    revision_blockers=REVISION_BLOCKER_LINTS,
+    stack_blockers=STACK_BLOCKER_LINTS,
+    revision_warnings=WARNING_LINTS,
+) -> TransplantAssessment:
+    """Build a `TransplantAssessment` by running landing lints.
+
+    Run each landing lint and append the result to the `TransplantAssessment`.
+    There are three categories of lint:
+        - `stack_blockers` are lints that inspect the entire state of the stack, and
+          will block landing the stack if the lint does not pass.
+        - `revision_blockers` are lints that inspect each individual revision and diff
+          pair, and will block landing the revision if the lint does not pass.
+        - `revision_warnings` are lints that inspect each individual revision and diff
+          pair, and will present a warning that must be acknowledged to land if the
+          lint does not pass.
+
+    Each type of lint takes the `TransplantAssessmentState` object, and the revision-level
+    blockers and warnings also take each `(revision, diff)` pair as arguments. Lints return
+    `None` on success, and a string reason explaining what went wrong in the lint on error.
+    """
+    assessment = TransplantAssessment()
+
+    # Run stack-level blocker checks.
+    for block in stack_blockers:
+        reason = block(transplant_state=transplant_state)
+        if reason is not None:
+            assessment.blockers.append(reason)
+
+    # Get the appropriate list of pairs to run checks against.
+    revision_check_pairs = transplant_state.revision_check_pairs()
+
+    # Run revision-level blockers checks.
+    for revision, diff in revision_check_pairs:
+        phid = revision["phid"]
+        for block in revision_blockers:
+            reason = block(
+                revision=revision,
+                diff=diff,
+                transplant_state=transplant_state,
+            )
+            if reason is not None:
+                assessment.blockers.append(reason)
+
+                if phid is not None and phid in transplant_state.landable_stack:
+                    transplant_state.landable_stack.remove_node(phid)
+                    transplant_state.stack.nodes[phid]["blocked"].append(reason)
+
+    # Run revision-level warning checks.
+    for revision, diff in revision_check_pairs:
+        for check in revision_warnings:
+            reason = check(
+                revision=revision, diff=diff, transplant_state=transplant_state
+            )
+
+            if reason is not None:
+                assessment.warnings.append(reason)
```

Comment:
nit (can be applied to above as well).

```suggestion
            if reason := check(
                revision=revision, diff=diff, transplant_state=transplant_state
            ) is not None:
                assessment.warnings.append(reason)
```


Another slightly simpler approach may be:
```python
results = [check(revision, diff, transplant_state) for check in revision_warnings]
assessment.warnings += [result for result in results if result is not None]
```